airflow-core/src/airflow/models/taskinstance.py

```diff
@@ -954,14 +954,15 @@ def next_retry_datetime(self):
         from airflow.sdk.definitions._internal.abstractoperator import MAX_RETRY_DELAY
 
         delay = self.task.retry_delay
-        if self.task.retry_exponential_backoff:
+        multiplier = self.task.retry_exponential_backoff if self.task.retry_exponential_backoff != 0 else 1.0
+        if multiplier != 1.0:
```

Comment:
```suggestion
        if multiplier != 1.0 and multiplier > 0:
```

To prevent critical failures, input validation or defensive checks should be implemented to ensure the exponential backoff multiplier is always greater than 0.